.github/workflows/tag.yml

```diff
@@ -0,0 +1,34 @@
+name: Set git tag
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 11 * * 4"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  set_tag:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: "main"
+          fetch-depth: 0
+          token: ${{secrets.PAT}}
+
+      - name: Set git tag
+        run: |
+          tag=$(date +'%Y%m%d')
+          git config --global user.name 'GitHub Actions'
+          git config --global user.email 'github-actions@users.noreply.github.com'
+
+          if git tag -l $tag; then
+            echo "Tag $tag already exists."
```

Comment:
we may want to fail the build in this case, so we get notifications, etc. when it occurs, since it should not ever happen.